### DISCUSSION TITLE
multiple methods for the same route

### DISCUSSION THREAD

**Discussion by mohad12211** [tokio-rs/axum#2050] [Q&A] [answered]
> # [Accepting multiple methods](https://github.com/tokio-rs/axum/blob/8374c3e789e71b4150fef46e274c9ed1ef9358bf/axum/src/docs/routing/route.md?plain=1#L62-L81)
> 
> To accept multiple methods for the same route you must add all handlers at the
> same time:
> 
> ```rust
> use axum::{Router, routing::{get, delete}, extract::Path};
> 
> let app = Router::new().route(
>     "/",
>     get(get_root).post(post_root).delete(delete_root),
> );
> 
> async fn get_root() {}
> 
> async fn post_root() {}
> 
> async fn delete_root() {}
> ```

the docs says **must** but this code works with no problems: 
> ```rust
>   let app: Router = Router::new()
>      .route("/", get(get_root))
>      .route("/", post(post_root))
>      .route("/", delete(delete_root));
>
> async fn get_root() {}
> 
> async fn post_root() {}
> 
> async fn delete_root() {}
> ```

I even added some code to the handlers and they work as expected, is it a coincidence? so in some cases it might not work?
or are the docs inaccurate?

**Comment by davidpdrsn** [comment, marked as answer]
Right, the docs are out of date here. Do you wanna submit a PR to make that more clear?

**Reply by mohad12211** [reply]
yeah, thank you for responding.
